test/uk/gov/hmrc/cgtpropertydisposals/controllers/homepage/FinancialDataControllerSpec.scala

```diff
@@ -0,0 +1,143 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.controllers.homepage
+
+import java.time.{LocalDate, LocalDateTime}
+
+import akka.stream.Materializer
+import cats.data.EitherT
+import cats.instances.future._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Headers, Result}
+import play.api.test.Helpers._
+import play.api.test.Helpers.CONTENT_TYPE
+import uk.gov.hmrc.cgtpropertydisposals.Fake
+import uk.gov.hmrc.cgtpropertydisposals.controllers.ControllerSpec
+import uk.gov.hmrc.cgtpropertydisposals.controllers.actions.AuthenticatedRequest
+import uk.gov.hmrc.cgtpropertydisposals.models.Error
+import uk.gov.hmrc.cgtpropertydisposals.models.Generators.sample
+import uk.gov.hmrc.cgtpropertydisposals.models.des.homepage.{FinancialDataRequest, FinancialDataResponse}
+import uk.gov.hmrc.cgtpropertydisposals.service.homepage.FinancialDataService
+import uk.gov.hmrc.cgtpropertydisposals.service.onboarding.AuditService
+import uk.gov.hmrc.http.HeaderCarrier
+import play.api.test.{FakeRequest, Helpers}
+import uk.gov.hmrc.cgtpropertydisposals.models.Generators._
+import uk.gov.hmrc.cgtpropertydisposals.models.ids.CgtReference
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class FinancialDataControllerSpec extends ControllerSpec {
+
+  val financialDataService = mock[FinancialDataService]
+  val auditService         = mock[AuditService]
+
+  val financialDataRequest = sample[FinancialDataRequest]
+
+  implicit val headerCarrier = HeaderCarrier()
+
+  def mockGetFinancialData(financialData: FinancialDataRequest)(response: Either[Error, FinancialDataResponse]) =
+    (financialDataService
+      .getFinancialData(_: FinancialDataRequest)(_: HeaderCarrier))
+      .expects(financialData, *)
+      .returning(EitherT.fromEither[Future](response))
+
+  implicit lazy val mat: Materializer = fakeApplication.materializer
+
+  val request = new AuthenticatedRequest(
+    Fake.user,
+    LocalDateTime.now(),
+    headerCarrier,
+    FakeRequest()
+  )
+
+  def fakeRequestWithJsonBody(body: JsValue) = request.withHeaders(Headers.apply(CONTENT_TYPE -> JSON)).withBody(body)
+
+  val controller = new FinancialDataController(
+    authenticate         = Fake.login(Fake.user, LocalDateTime.of(2020, 1, 1, 15, 47, 20)),
+    financialDataService = financialDataService,
+    auditService         = auditService,
+    cc                   = Helpers.stubControllerComponents()
+  )
+
+  def getFinancialDataRequest(r: FinancialDataRequest) =
```

Comment:
this is a bit confusing - you're taking in a `FinancialDataReqest` and spitting out a `FinancialDataRequest`. It'd be better if you had a correct `FinancialDataRequest` in the first place and not need this method